libcxx/test/support/test_macros.h

```diff
@@ -10,6 +10,8 @@
 #ifndef SUPPORT_TEST_MACROS_HPP
 #define SUPPORT_TEST_MACROS_HPP
 
+#pragma clang diagnostic warning "-Wdangling"
```

Comment:
What's this for?

Reply:
Do you have more details about what exactly was dangling? That seems suspicious to me, since we should never have UB in our tests. It would be possible to demote this to a warning, but I don't think we want to -- IMO the test suite did the right thing by making this an error cause we should probably be fixing it.

Reply:
Was it that part in `libcxx/test/std/containers/views/views.span/span.cons/array.pass.cpp`:

```c++
    std::span<const int> s5 = {{1,2}};
#if TEST_STD_VER >= 26
    std::span<const int, 2> s6({1, 2});
#else
    std::span<const int, 2> s6 = {{1,2}};
#endif
    assert(s5.size() == 2);  // and it dangles
    assert(s6.size() == 2);  // and it dangles
```

I don't understand how that would trigger `-Wdangling` after this patch though?